alpha-build-machinery/make/examples/multiple-binaries/Makefile.test

```diff
@@ -1,22 +1,38 @@
 SHELL :=/bin/bash -euo pipefail
 
 all:
-	$(MAKE) -C . build
+	$(MAKE) build
 	[[ -f ./openshift ]]
 	[[ -f ./oc ]]
 
 	# test version is set correctly when linking
 	#   majorFromGit, minorFromGit are deprecated upstream and set to empty value
 	#   we avoid comparing time to avoid flakes
-	diff <( ./oc | sed '$$d' ) <( \
+	#   we avoid comparing git tree state
+	diff <( ./oc | grep -v -e 'clean' -e 'dirty' | sed '$$d' ) <( \
 		echo ""; \
 		echo ""; \
 		git rev-parse --short "HEAD^{commit}" 2>/dev/null; \
 		git describe --long --tags --abbrev=7 --match 'v[0-9]*' || echo 'v0.0.0-unknown'; \
-		git diff --quiet && echo 'clean' || echo 'dirty'; \
 	)
 
-	$(MAKE) -C . clean
+	$(MAKE) clean
 	[[ ! -f ./openshift ]]
 	[[ ! -f ./oc ]]
+
+	[[ ! -f ./_bin ]]
+	$(MAKE) cross-build
+	[[ ! -f ./openshift ]]
+	[[ ! -f ./oc ]]
+	[[ -f ./_bin/darwin_amd64/openshift ]]
+	[[ -f ./_bin/darwin_amd64/oc ]]
+	[[ -f ./_bin/windows_amd64/openshift.exe ]]
+	[[ -f ./_bin/windows_amd64/oc.exe ]]
```

Comment:
makefiles testing makefiles 😱

Reply:
its shell in the end but we can split it into multiple targets to be able to run a specific test